src/module-loader/dotnet/src/MorganStanley.ComposeUI.ModuleLoader.Abstractions/IModuleRunner.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Morgan Stanley makes this available to you under the Apache License,
+// Version 2.0 (the "License"). You may obtain a copy of the License at
+// 
+//      http://www.apache.org/licenses/LICENSE-2.0.
+// 
+// See the NOTICE file distributed with this work for additional information
+// regarding copyright ownership. Unless required by applicable law or agreed
+// to in writing, software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions
+// and limitations under the License.
+
+namespace MorganStanley.ComposeUI.ModuleLoader;
+
+public interface IModuleRunner
+{
+    string ModuleType { get; }
+
+    Task Start(IModuleInstance moduleInstance, StartupContext startupContext, Func<Task> pipeline);
+
+    Task Stop(IModuleInstance moduleInstance, CancellationToken cancellationToken = default);
```

Comment:
Do we need cancellation here? I'm not sure it's feasible to cancel, other than from the module itself (we'll need some mechanism to do that, but it's not by a cancellation token)

Reply:
I don't know if we need cancellation here but I agree we don't need cancellation here *right now* :) I'll remove that argument.

Reply:
It's just that if you cancel it halfway, what's the expected behavior?